src/producer.rs

```diff
@@ -17,12 +17,23 @@ pub enum ArchiveType {
     Dir(PathBuf),
 }
 
+pub enum FilePath<'a> {
```

Comment:
We could probably get rid of this, if we make is_gcno_llvm accept a generic reader (e.g. like `parse_lcov` in src/parser.rs).
It's a minor thing though, so I've merged this and I'll file a follow-up 'good-first-bug' issue for it.

Reply:
is_gcno_llvm already take a std::io::Read, here I need to make the diff between Zip and Dir finally, since we've a Read when we're in a zip and we've a file when we're in a dir.

Reply:
The file could be read at https://github.com/mozilla/grcov/pull/137/files#diff-55015d4ff29cae5f493ea6ded47589b7R136, and the buffer passed to handle_file, so that in both cases (FilePath::File and FilePath::Path) you'd always pass a buffer to handle_file